spinoso-time/src/time/tzrs/offset.rs

```diff
@@ -100,38 +108,41 @@ impl<'a> Offset {
         let offset_name = offset_hhmm_from_seconds(offset);
         let local_time_type =
             LocalTimeType::new(offset, false, Some(offset_name.as_bytes())).expect("Couldn't create fixed offset");
-        Self::Fixed([local_time_type])
+        Self {
+            inner: OffsetType::Fixed(local_time_type),
+        }
     }
 
     /// Generate an offset based on a provided [`tz::timezone::TimeZoneRef`].
     ///
     /// This can be combined with [`tzdb`] to generate offsets based on
     /// predefined IANA time zones.
     ///
-    /// ```
-    /// use spinoso_time::tzrs::Offset;
-    /// use tzdb::time_zone::pacific::AUCKLAND;
-    /// let offset = Offset::tz(AUCKLAND);
-    /// ```
+    /// [`tz:timezone::TimeZoneRef`]: https://docs.rs/tz-rs/0.6.9/tz/timezone/struct.TimeZoneRef.html
+    /// [`tzdb`]: https://docs.rs/tzdb/latest/tzdb/index.html
     #[inline]
     #[must_use]
-    pub fn tz(tz: TimeZoneRef<'static>) -> Self {
-        Self::Tz(tz)
+    pub(crate) fn tz(tz: TimeZoneRef<'static>) -> Self {
+        Self {
+            inner: OffsetType::Tz(tz),
+        }
     }
 
     /// Returns a `TimeZoneRef` which can be used to generate and project
     /// _time_.
     #[inline]
     #[must_use]
-    pub fn time_zone_ref(&self) -> TimeZoneRef<'_> {
-        match self {
-            Self::Utc => TimeZoneRef::utc(),
-            Self::Fixed(local_time_types) => match TimeZoneRef::new(&[], local_time_types, &[], &None) {
-                Ok(tz) => tz,
-                Err(_) => GMT,
-            },
-
-            Self::Tz(zone) => *zone,
+    pub(crate) fn time_zone_ref(&self) -> TimeZoneRef<'_> {
+        match self.inner {
+            OffsetType::Utc => TimeZoneRef::utc(),
+            OffsetType::Fixed(ref local_time_type) => {
+                match TimeZoneRef::new(&[], std::slice::from_ref(local_time_type), &[], &None) {
```

Comment:
can you `use std::slice;` at the top of this file and change this to:

```suggestion
                match TimeZoneRef::new(&[], slice::from_ref(local_time_type), &[], &None) {
```

this codebase tends not to use fully qualified paths into `std` and `core`.